specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-09-10-preview/HybridCompute.json

```
@@ -1635,6 +1635,201 @@
         }
       }
     },
+    "/providers/Microsoft.HybridCompute/locations/{location}/publishers/{publisher}/extensionTypes/{extensionType}/versions/{version}": {
```

```suggestion
    "/subscriptions/{subscriptionId}/providers/Microsoft.HybridCompute/locations/{location}/publishers/{publisher}/extensionTypes/{extensionType}/versions/{version}": {
```

@jiaqi-beep: Should this path include the subscription prefix, like the similarly-named operation above?

https://github.com/Azure/azure-rest-api-specs/pull/30656/files#diff-228c58b2c8d013d70a88a9cc262d060822288f28f91b63b4908b34527d4e6ee8L1514-L1519

---

No. V2 API that I am adding in this PR removes the need for subscription ID. Automated check seems to be confused that paths with and without subscription id are different. fyi this question has been asked by previous reviewers.

---

This is good information to put in the suppression in your `readme.md`, instead of just "false positive", so everyone reviewing your spec (in any PR, in any branch) can undersand this, without needing to ask again.

---

If your spec is correct, and the tool is incorrect to compare paths with and without subscription ID, I will open an issue so the tool can be fixed.

---

Thanks

---

Actually, the check gives the wrong error, but it is true that my response model in the GET collection operation "ExtensionMetadataV2_List" does not match with the response model in the individual GET operation "ExtensionMetadataV2_Get", see ExtensionMetadataV2_List.json and ExtensionMetadataV2_Get.json.

And this is by design, since GET should have more additional properties than LIST.

---

In this case, I think there are two issues:

1. Your spec is violating the **spirit** of rule `GetCollectionResponseSchema`, which is "collection APIs should return the same model as individual GETs".  You say this is by design, but it goes directly against the ARM guidelines.  The reason you are choosing to violate this guideline should be documented in your readme suppression.

2. Rule `GetCollectionResponseSchema` may have one or more bugs, related to detecting and/or reporting errors in APIs that vary only by the subscription prefix.  I will investigate this, and open issues in https://github.com/Azure/azure-openapi-validator.  You can link to these issues from your readme suppression.

---

> Actually, the check gives the wrong error, but it is true that my response model in the GET collection operation "ExtensionMetadataV2_List" does not match with the response model in the individual GET operation "ExtensionMetadataV2_Get", see ExtensionMetadataV2_List.json and ExtensionMetadataV2_Get.json.
> 
> And this is by design, since GET should have more additional properties than LIST.

This breaks ARM design rules. Both payloads should return the same resource payload. If you want to support a list operation that returns subset objects, you need to do that with a filter parameter to the LIST method. This is very important, otherwise core ARM features (Azure Policy, Azure Resource Graph, maybe others) will be broken, which is a poor experience for customers.

---

@mentat9  I've modified the spec so that the collection and individual GET operations have matching response